File: examples/python/read_xlsx.py
"""Load an XLSX and print its IR sections as simple tables."""
import sys
from office_oxide import Document


def main(path: str) -> None:
    with Document.open(path) as doc:
        ir = doc.to_ir()
        for i, section in enumerate(ir["sections"]):
            print(f"# sheet {i}: {section.get('title')}")
            for el in section["elements"]:
                if el["type"] == "table":
                    for row in el["rows"]:
                        cells = [c.get("text", "") for c in row["cells"]]
                        print("\t".join(cells))


if __name__ == "__main__":
    if len(sys.argv) != 2:
        sys.exit("usage: read_xlsx.py <file.xlsx>")
    main(sys.argv[1])

File: examples/javascript/read_xlsx.mjs
#!/usr/bin/env node
import { Document } from 'office-oxide';

if (process.argv.length !== 3) {
  console.error('usage: read_xlsx.mjs <file.xlsx>');
  process.exit(1);
}

const doc = Document.open(process.argv[2]);
try {
  const ir = doc.toIr();
  for (let i = 0; i < ir.sections.length; i++) {
    const s = ir.sections[i];
    console.log(`# sheet ${i}: ${s.title ?? ''}`);
    for (const el of s.elements) {
      if (el.type !== 'table') continue;
      for (const row of el.rows) {
        console.log(row.cells.map(c => c.text ?? '').join('\t'));
      }
    }
  }
} finally {
  doc.close();
}
